packages/halo/src/react/index.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  names as themeNames,
+  useTheme
+} from '@pluralsight/ps-design-system-theme'
+import {
+  RefForwardingComponent,
+  ValueOf,
+  canUseDOM
+} from '@pluralsight/ps-design-system-util'
+import { StyleAttribute, compose, css } from 'glamor'
+import polyfillFocusWithin from 'focus-within'
+import React, { HTMLAttributes, forwardRef } from 'react'
+
+import stylesheet, { BASE_CLASSNAME, themeClasses } from '../css'
+import * as vars from '../vars'
+
+if (canUseDOM()) polyfillFocusWithin(document)
+
+export type StyleFn<P> = (
+  themeName: ValueOf<typeof themeNames>,
+  props: P
```

Comment:
so previously this was a little different then the component props because the values were required(the components are optional)

Reply:
oh, snap, lemme take another look...